test/es6/regex-symbols.js

```diff
@@ -600,9 +663,184 @@ var tests = [
+    },
+    {
+        name: "RegExp.prototype[@@split] should run the built-in 'split' when none of the observable properties are overridden",
+        body: function () {
+            var pattern = "-";
+            var input = "-a-b--c-";
+
+            function verify(assertMessagePrefix, re) {
+                var result = re[Symbol.split](input);
+
+                assert.areEqual(6, result.length, assertMessagePrefix + ": result.length");
+                assert.areEqual("", result[0], assertMessagePrefix + ": result[0]");
+                assert.areEqual("a", result[1], assertMessagePrefix + ": result[1]");
+                assert.areEqual("b", result[2], assertMessagePrefix + ": result[2]");
+                assert.areEqual("", result[3], assertMessagePrefix + ": result[3]");
+                assert.areEqual("c", result[4], assertMessagePrefix + ": result[4]");
+                assert.areEqual("", result[5], assertMessagePrefix + ": result[5]");
+            }
+
+            verify("non-sticky", new RegExp(pattern));
+            verify("sticky", new RegExp(pattern, "y"));
+        }
+    },
+    {
+        name: "RegExp.prototype[@@split] should 'Get' 'flags' when it is overridden",
+        body: function () {
+            var re = /-/;
+
+            var calledFlags = false;
+            var getFlags = function () {
+                calledFlags = true;
+                return "";
+            };
+            Object.defineProperty(re, "flags", {get: getFlags});
+
+            re[Symbol.split]("");
+
+            assert.isTrue(calledFlags, "'flags' getter is called");
+        }
+    },
+    {
+        name: "RegExp.prototype[@@split] should construct a new RegExp using Symbol.species",
+        body: function () {
+            var re = /./i;
+
+            var ctorCalled = false;
+            var ctorThis = undefined;
+            var ctorArguments = undefined;
+            var ctorResult = /different regexp/i
+            re.constructor = function () {}
+            re.constructor[Symbol.species] = function () {
+                ctorCalled = true;
+                ctorThis = this;
+                ctorArguments = arguments;
+
+                return ctorResult;
+            }
+
+            re[Symbol.split]("123");
+
+            assert.isTrue(ctorCalled, "constructor is called");
+            assert.areEqual(re, ctorArguments[0], "constructor is passed the original RegExp object");
+            assert.areEqual("iy", ctorArguments[1], "constructor is passed the correct flags (including 'y')");
+        }
+    },
+    {
+        name: "RegExp.prototype[@@split] should return an empty Array when the input size is 0 and the RegExp matches the empty string",
+        body: function () {
+            function assertResult(result) {
+                assert.areEqual(0, result.length, "result.length");
+            }
+            var re = /(?:)/;
+            var input = "";
+            verifySymbolSplitResult(assertResult, re, input);
```

Comment:
Ok I meant new RegExp("")[Symbol.split](""), which is the same above case.

---
In reply to: [54801523](https://github.com/Microsoft/ChakraCore/pull/416#discussion_r54801523) [](ancestors = 54801523)